src/src/com/microsoft/aad/adal/AuthenticationContext.java

```diff
@@ -1453,12 +1421,12 @@ private AuthenticationResult localFlow(CallbackHandler callbackHandle,
                         new InstrumentationPropertiesBuilder(request, authResult).build());
                 return authResult;
             }
-        } 
+        }
         
         // refresh token does not exist or refresh token request failed to give back the token. 
         // If it's non-silent request, will try to acquire token interactively. 
         // if it's silent request, will return the AUTH_REFRESH_FAILED_PROMPT_NOT_ALLOWED.
-        if (refreshItem == null || authResult == null 
+        if (authResult == null 
                 || (authResult != null && StringExtensions.IsNullOrBlank(authResult.getAccessToken()))) {
             Logger.v(TAG, "Refresh token is not available or refresh token request failed to return token.");
```

Comment:
Done. 

---
In reply to: [62959464](https://github.com/AzureAD/azure-activedirectory-library-for-android/pull/611#discussion_r62959464) [](ancestors = 62959464)